simtools/corsika/corsika_output.py

```diff
@@ -0,0 +1,1002 @@
+        Input file (IACT file) provided by the CORSIKA simulation.
+    Raises
+    ------
+    FileNotFoundError:
+        if the input file given does not exist.
+    """
+
+    def __init__(self, input_file):
+
+        self._logger = logging.getLogger(__name__)
+        self._logger.debug("Init CorsikaOutput")
+        self.input_file = input_file
+
+        if not isinstance(self.input_file, Path):
+            self.input_file = Path(self.input_file)
+        if not self.input_file.exists():
+            raise FileNotFoundError
+
+        self._tel_positions = None
+
+        self._allowed_histograms = {"hist_position", "hist_direction", "hist_time_altitude"}
+        self._allowed_1D_labels = {"wavelength", "time", "altitude"}
+        self._allowed_2D_labels = {"counts", "density", "direction", "time_altitude"}
+        self._events_information = None
+        self._get_event_information()
+
+    @property
+    def telescope_indices(self):
+        """
+        The telescope index (or indices) as a class attribute.
+        """
+        return self._telescope_indices
+
+    @telescope_indices.setter
+    def telescope_indices(self, telescope_new_indices):
+        """
+        Set the telescope index (or indices) as a class attribute.
+
+        Parameters
+        ----------
+        telescope_new_indices: int or list of int
+            The indices of the specific telescopes to be inspected. If not specified, all telescopes
+            are treated together in one histogram.
+
+        Raises
+        ------
+        TypeError:
+            if the indices passed through telescope_index are not of type int.
+        """
+        if telescope_new_indices is not None:
+            if not isinstance(telescope_new_indices, list):
+                telescope_new_indices = [telescope_new_indices]
+            for one_telescope in telescope_new_indices:
+                if not isinstance(one_telescope, int):
+                    msg = "The index or indices given are not of type int."
+                    self._logger.error(msg)
+                    raise TypeError
+        self._telescope_indices = telescope_new_indices
+
+    def _get_directive_cosinus(self, zenith_angle, azimuth_angle):
+        """
+        Get the direction cosinus (X and Y) from the incoming particle, which helps defining the
+        range of the histograms built from the photon incoming directions.
+
+        Parameters
+        ----------
+        zenith_angle: astropy.units.rad
+            Zenith angle (in radians) of the observations (in the CORSIKA coordinate system).
+            It is used to rotate the observation plane (in zenith) of the telescope and to plot
+            the sky map of the incoming direction of photons.
+        azimuth_angle: astropy.units.rad
+            Azimuth angle of observation (in radians).
+            See above for more details.
+
+        Returns
+        -------
+        np.array:
+            Directive cosinus in the X direction.
```

Comment:
You can see this sketch for the definition below. We actually dont know the angles between the incoming direction and individual axis but we do know the zenith and azimuth angles, from which we calculate the direction cosinus directly, without explicitly deriving the angles. In any case, we dont need this function anymore. I was trying to plot the azimuth x zenith incoming direction before but it turned out that it would not be strictly correct because of the signal of the angle. In some cases I would not be able to tell if the angle is positive or negative, since I only have the cosinus of the angle. Therefore, I have abandoned the idea and I will then now also remove this function, unless you have something against it?
![image](https://user-images.githubusercontent.com/50831535/231480597-23342d6a-6e33-4f65-b709-6ae46cdcabbd.png)

Reply:
Nothing against removing unnecessary code